dbt_schema_builder/schema.py

```diff
@@ -3,45 +3,71 @@
     """
 
-    def __init__(self, schema_name, exclusion_list, inclusion_list, relations=None):
+    def __init__(self, schema_name, exclusion_list, inclusion_list, soft_delete_column_name,
+                 soft_delete_sql_predicate, relations=None):
         self.schema_name = schema_name
         self.exclusion_list = exclusion_list
         self.inclusion_list = inclusion_list
+        self.soft_delete_column_name = soft_delete_column_name
+        self.soft_delete_sql_predicate = soft_delete_sql_predicate
         self.relations = relations
 
+        self.validate()
+
     def __repr__(self):
         return self.schema_name
 
+    def validate(self):
+        """
+        Confirm that the class is instantiated correctly.
+        """
+        if self.soft_delete_column_name is not None:
+            if self.soft_delete_sql_predicate in (None, '') or not isinstance(self.soft_delete_sql_predicate, str):
+                raise InvalidConfigurationException(
+                    'Schema {} has an invalid SOFT_DELETE configuration. '
+                    'SOFT_DELETE must be a single dict with the column name to look for and '
+                    'the SQL needed to exclude the soft deleted rows. '.format(self.schema_name)
+                )
+
+        if self.exclusion_list and self.inclusion_list:
+            raise InvalidConfigurationException(
+                'Schema {} has both INCLUDE and EXCLUDE sections in its'
+                'sections in its configuration file'.format(self.schema_name)
+            )
+
     @classmethod
     def from_config(cls, schema_name, config):
         """
         Construct a Schema object from a config dictionary. This is encapuslated
         into it's own function to help declutter the `run` function in
         builder.py.
-        TODO: add code to parse out soft delete and unmanaged table configs here
+        TODO: add code to parse out un-managed table configs here
         """
+        exclusion_list = []
+        inclusion_list = []
+        soft_delete_column_name = None
+        soft_delete_sql_predicate = None
+
         if config:
-            exclusion_list = config['EXCLUDE'] if config.get('EXCLUDE') else []
-            inclusion_list = config['INCLUDE'] if config.get('INCLUDE') else []
-        else:
-            exclusion_list = []
-            inclusion_list = []
+            exclusion_list = config.get('EXCLUDE', [])
+            inclusion_list = config.get('INCLUDE', [])
 
-        if exclusion_list and inclusion_list:
-            raise InvalidConfigurationException(
-                'Schema {} has both INCLUDE and EXCLUDE sections in its'
-                'sections in its configuration file'.format(schema_name)
-            )
+            if 'SOFT_DELETE' in config:
+                for k, v in config['SOFT_DELETE'].items():
+                    soft_delete_column_name = k
+                    soft_delete_sql_predicate = v
```

Comment:
Is the existence of only a single key/value pair checked elsewhere? Because this code will simply take the "last" key/value pair it encounters from the SOFT_DELETE section to use below.